code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -2,32 +2,153 @@
 	density = 1
 	anchored = 1
 	icon = 'icons/obj/cult.dmi'
+	var/cooldowntime = 0
+	var/health = 100
+	var/maxhealth = 100
 
+/obj/structure/cult/proc/getETA()
+    var/time = round((0-(world.time-cooldowntime))/600, 1)
+    var/eta = "[time] minutes."
+    if(time == 1)
+        eta = "about one minute."
+    else if(time == 0)
+        eta = "less than thirty seconds."
+    return eta
+         
 /obj/structure/cult/talisman
 	name = "altar"
-	desc = "A bloodstained altar dedicated to Nar-Sie"
+	desc = "A bloodstained altar dedicated to Nar-Sie."
 	icon_state = "talismanaltar"
 
+	
+/obj/structure/cult/talisman/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You don't even begin to understand what these words mean...</span>"
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	cooldowntime = world.time + 2400
+	spawn(2400)
```

Comment:
It doesn't actually work like that. I'm trying to determine if the spawn is needed at all actually.